### PR TITLE
Add Retry-After hint

### PR DESCRIPTION
**Pull request by gautam1168**
Fixes issue [1878](https://github.com/PostgREST/postgrest/issues/1878)

### CODE REVIEW

**Comment by steve-chavez** [comment]
Great work @gautam1168!

I've just tried your change by doing

```bash
sudo systemctl stop postgresql

curl -I localhost:3000/projects

HTTP/1.1 503 Service Unavailable

Date: Tue, 17 Aug 2021 16:54:18 GMT
Server: postgrest/8.0.0 (b3899e7)
Retry-After: 16
Content-Type: application/json; charset=utf-8
```

`Retry-After` is now showing!

Ideally we'd be able to add a test, but our `io-tests`(in Python) need some work to support restarting/starting the postgres instance(https://github.com/PostgREST/postgrest/issues/1766).

So this should be good. Don't forget to add an entry to the [CHANGELOG](https://github.com/PostgREST/postgrest/blob/main/CHANGELOG.md#unreleased)!

**Comment by steve-chavez** [comment]
Just noted something that perhaps should be handled in another PR: the connection worker delay is not being respected.

On v7, it used to be that all requests would get rejected with 503 while the connection worker is running. But on v8.0, the connection recovery is done whenever a new request comes in:

```bash
curl -I localhost:3000/projects
HTTP/1.1 503 Service Unavailable
Date: Tue, 17 Aug 2021 17:15:12 GMT
Server: postgrest/8.0.0 (b3899e7)
Retry-After: 32
Content-Type: application/json; charset=utf-8

sudo systemctl start postgresql

## just a second after

curl -I localhost:3000/projects
HTTP/1.1 200 OK
Date: Tue, 17 Aug 2021 17:15:46 GMT
Server: postgrest/8.0.0 (b3899e7)
Content-Range: 0-58/*
Content-Location: /projects
Content-Type: application/json; charset=utf-8

## postgrest is already recovered

## The connection worker will report the reconnection later(after 32 sec):
## 17/Aug/2021:12:16:05 -0500: Connection successful
## 17/Aug/2021:12:16:05 -0500: Config re-loaded
## 17/Aug/2021:12:16:06 -0500: Schema cache loade
```

This is probably a side effect of doing https://github.com/PostgREST/postgrest/pull/1559.

**Comment by gautam1168** [comment]
@steve-chavez  I have added an entry to CHANGELOG.md in the unreleased fixes section. 

Regarding connection worker delay not being respected. I am assuming you will be opening another issue to handle that and maybe I will give that a try too. But next want to try to add the limit=0 feature request from another issue.

**Comment by gautam1168** [comment]
I figured since its not a bug it should be in the `Added` section and moved it.

I will figure out how to rebase against the upstream repository and resolve this conflict.

**Comment by gautam1168** [comment]
@steve-chavez I have resolved the conflict and squashed my changes to a single commit. I have also added a better commit message and included the issue number that is fixed in the description.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
  - #1783, Include partitioned tables into the schema cache. Allows embedding, UPSERT, INSERT with Location response, OPTIONS request and OpenAPI support for partitioned tables - @laurenceisla
 
+ - #1878, Add Retry-After hint when in recovery mode
+
```
Comment (monacoremo) [inline CHANGELOG.md:13]:
```suggestion
 - #1878, Add Retry-After hint header when in recovery mode - @gautam1168
 
```

**Review by monacoremo — APPROVED** [review]
Great work!

**Review by steve-chavez — APPROVED** [review]
LGTM! Merging :rocket:

**Pull request merged**